AdobeStockImage/Model/SaveImagePreview.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeStockImage\Model;
+
+use Magento\AdobeStockAsset\Model\SavePreviewImageAssetStrategy;
+use Magento\AdobeStockAssetApi\Api\Data\AssetInterface;
+use Magento\AdobeStockImage\Model\IsImageValidToSaveCondition\IsImageExistsConditionChain;
+use Magento\AdobeStockImageApi\Api\SaveImagePreviewInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class SaveImagePreview
+ */
+class SaveImagePreview implements SaveImagePreviewInterface
+{
+    /**
+     * @var GetImage
+     */
+    private $getImage;
+
+    /**
+     * @var IsImageExistsConditionChain
+     */
+    private $isMediaExistsConditionChain;
+
+    /**
+     * @var SavePreviewImageAssetStrategy
+     */
+    private $savePreviewImageAssetStrategy;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * SaveImagePreview constructor.
+     *
+     * @param GetImage                    $getImage
+     * @param IsImageExistsConditionChain $isMediaExistsConditionChain
+     * @param SavePreviewImageAssetStrategy    $savePreviewImageAssetStrategy
+     * @param LoggerInterface             $logger
+     */
+    public function __construct(
+        GetImage $getImage,
+        IsImageExistsConditionChain $isMediaExistsConditionChain,
+        SavePreviewImageAssetStrategy $savePreviewImageAssetStrategy,
+        LoggerInterface $logger
+    ) {
+        $this->getImage = $getImage;
+        $this->isMediaExistsConditionChain = $isMediaExistsConditionChain;
+        $this->savePreviewImageAssetStrategy = $savePreviewImageAssetStrategy;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(int $mediaId, string $destinationPath): bool
+    {
+        try {
+            $searchResult = $this->getImage->execute($mediaId);
+            if (!$this->isMediaExistsConditionChain->execute($searchResult)) {
```

Comment:
I don't think it is necessary to introduce a new class
```suggestion
            if (!$searchResults->getTotalCount()) {
```

Reply:
I added it as an entry point of possible condition extensibility and reuse for a licensed image. Should remove it?